### ISSUE TITLE
Loopcast index occasionally skips 1

### ISSUE THREAD

**Issue by Hubry** [Vazkii/Psi#596]
![obraz](https://user-images.githubusercontent.com/26120076/83892566-226c4f80-a74f-11ea-8130-e123c9bc6e8a.png)
Repro spell is just channeled debug + loopcast index into number, time into target.
Second loop happens 1 tick earlier than expected and skips index 1.  This does not happen consistently, but quite often.
Versions are whatever we have on the patreon server.

Spell I found this out with: 
<details>
  <summary>Click to expand</summary>

```json
{modsRequired:[{modVersion:"r1.2-79",modName:"psi"}],spellName:"Single Pure Daisy",uuidMost:-2952801170173113213L,validSpell:1b,spellList:[{data:{params:{_target:4},key:"psi:trick_die"},x:2,y:3},{data:{key:"psi:constant_number",constantValue:"4"},x:3,y:2},{data:{params:{_number2:1,_number3:0,_number1:4},key:"psi:operator_subtract"},x:3,y:3},{data:{key:"psi:constant_number",constantValue:"3"},x:3,y:4},{data:{key:"psi:constant_number",constantValue:"9"},x:4,y:1},{data:{params:{_number2:1,_number3:0,_number1:2},key:"psi:operator_subtract"},x:4,y:2},{data:{key:"psi:selector_loopcast_index"},x:4,y:3},{data:{params:{_number2:3,_number3:0,_number1:1},key:"psi:operator_integer_divide"},x:4,y:4},{data:{key:"psi:selector_ruler_vector"},x:4,y:5},{data:{params:{_number:3},key:"psi:trick_break_loop"},x:5,y:2},{data:{params:{_number2:4,_number1:3},key:"psi:operator_modulus"},x:5,y:3},{data:{params:{_x:1,_y:0,_z:3},key:"psi:operator_vector_construct"},x:5,y:4},{data:{params:{_vector3:0,_vector2:3,_vector1:1},key:"psi:operator_vector_sum"},x:5,y:5},{data:{params:{_direction:0,_position:1},key:"psi:trick_place_block"},x:5,y:6},{data:{key:"psi:constant_number",constantValue:"3"},x:6,y:3},{data:{params:{_position:3},key:"psi:trick_break_block"},x:6,y:5}],uuidLeast:-8931869920724335846L}
```

</details>

**Comment by Math321** [comment]
Any word on this getting fixed?

**Comment by Kamefrede** [comment]
i think i have this fixed on 1.16

**Comment by Kamefrede** [comment]
Closing as it seems to be fixed on 1.16, will reopen if it pops up again
